Storage/Bookmarks/BookmarksModel.swift

```diff
@@ -296,6 +296,47 @@ open class PrependedBookmarkFolder: BookmarkFolder {
     }
 }
 
+open class ConcatenatedBookmarkFolder: BookmarkFolder {
+    fileprivate let main: BookmarkFolder
+    fileprivate let prepend: BookmarkFolder
+
+    init(main: BookmarkFolder, prepend: BookmarkFolder) {
+        self.main = main
+        self.prepend = prepend
+        super.init(guid: main.guid, title: main.guid)
+    }
+
+    override open var count: Int {
+        return main.count + prepend.count
+    }
+
+    override open subscript(index: Int) -> BookmarkNode? {
+        if index < prepend.count {
+            return prepend[index]
+        } else {
+            let adjustedIndex = index - prepend.count
+            return main[adjustedIndex]
+        }
+    }
+
+    override open func itemIsEditableAtIndex(_ index: Int) -> Bool {
+        if index < prepend.count {
+            return prepend.itemIsEditableAtIndex(index)
+        } else {
+            let adjustedIndex = index - prepend.count
+            return prepend.itemIsEditableAtIndex(adjustedIndex)
+        }
+    }
+
+    override open func removeItemWithGUID(_ guid: GUID) -> BookmarkFolder? {
+        guard let removedFolder = main.removeItemWithGUID(guid) else {
```

Comment:
If you fix `removeItemWithGUID` to never return `nil` — it no longer needs to — then a safe thing for you to do is to just do:

```
return ConcatenatedBookmarkFolder(main: main.removeItemWithGUID(guid), prepend: prepend.removeItemWithGUID(guid))
```